packages/thirdweb/src/react/core/hooks/auth/useSiweAuth.ts

```diff
@@ -56,6 +57,13 @@ export function useSiweAuth(
 
   const queryClient = useQueryClient();
 
+  // We need to reset the react query cache on mount for components like ConnectEmbed that are unmount and remount depending on the auth / connection state
```

Comment:
I would expect useQuery to already refetch on mount? That is the standard behavior.